pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -311,19 +316,44 @@ func (s *Server) initInprocessAnalysisController(args *PilotArgs) error {
 	processing := components.NewProcessing(processingArgs)
 
 	s.addStartFunc(func(stop <-chan struct{}) error {
-		if err := processing.Start(); err != nil {
-			return err
-		}
+		go leaderelection.
+			NewLeaderElection(args.Namespace, args.PodName, leaderelection.StatusController, s.kubeClient).
```

Comment:
How will this work with multi revision ? If you have a master across all revisions - most likely the updates will be wrong if you pick a canary for example. 

If it's across each revision - you may have overrides and conflicts, since one revision may set a status while another a different one.

Most resources will be visible and sent in all CP revisions.

Reply:
I'm not clear on what it means to "have a master across all revisions".  Because we are using leader election, there should only be one status leader at a time, and that leader will take input from any followers publishing DistributionReports.

Reply:
And that master will be which revision ? Canary ? A test revision with no workloads ? 
And worse - if the master happens to be the canary where you're testing some crazy config - will all the statuses revert to something bad because the canary tacking the status gets NACKs ( while the stable prod config is valid ) ?

Reply:
I.e there is no way in master election to guarantee the master will be the right one.

At least in k8s - other kind of elections are different :-)

Reply:
I will add a section in the documentation indicating that status should only be enabled on master control planes.   This will prevent canaries from contributing to status.

Reply:
Shouldn't it be the other way around ? A change should be done first on the canary - if there is an error that's where it should be first found. Once it gets to the master - it's too late.

Reply:
Also this reminds me - there is no flag or env to turn it off it seems ? Or is it tied to the flag that disables analysis controller ? Is that off by default as well ?

Reply:
You can find the enable/disable flags here: https://github.com/istio/istio/pull/22873/files#diff-850f68f22b775c6d7475a839ca5b382f

This change is really not meant to allow or support canary rollout - but it might be repurposed that way.  Let's set up a design discussion regarding how we could support this in 1.7.